### PR TITLE
Check Gradle compatibility when importing fails

### PR DESCRIPTION
**Pull request by CsCherrYY**
compatibility

requires https://github.com/eclipse/eclipse.jdt.ls/pull/1965

Signed-off-by: Shi Chen <chenshi@microsoft.com>

### CODE REVIEW

File: src/standardLanguageClient.ts [hunk 1]
```diff
@@ -144,6 +150,44 @@ export class StandardLanguageClient {
...
+									jdkHomeEnvRuntimes.push(javaRuntime);
+								} else {
+									runtimes.push(javaRuntime);
+								}
+							}
+						}
+						let targetRuntime;
+						if (pathEnvRuntimes.length > 0) {
+							targetRuntime = this.findHighestRuntime(pathEnvRuntimes);
+						} else if (javaHomeEnvRuntimes.length > 0) {
```
Comment (testforstephen) [inline src/standardLanguageClient.ts:177]: You wrote it too complicated. Can you reference 
https://github.com/redhat-developer/vscode-java/blob/562afa80c0f705613c6ed3148b4d9eac82093b26/src/requirements.ts#L150-L161

**Review by testforstephen — APPROVED** [review]

File: src/standardLanguageClient.ts [hunk 2]
```diff
@@ -217,6 +240,44 @@ export class StandardLanguageClient {
...
 	}
 
+	private showGradleCompatibilityIssueNotification(message: string, options: string[], projectUri: string, newJavaHome: string) {
+		window.showErrorMessage(message + " [Learn More](https://docs.gradle.org/current/userguide/compatibility.html)", ...options).then(async (choice) => {
+			if (choice === GET_JDK) {
+				commands.executeCommand(Commands.OPEN_BROWSER, Uri.parse(getJdkUrl()));
+			} else if (choice.startsWith(USE_JAVA)) {
+				await workspace.getConfiguration().update(GRADLE_IMPORT_JVM, newJavaHome, ConfigurationTarget.Global);
+				commands.executeCommand("workbench.action.openSettings", GRADLE_IMPORT_JVM);
+				commands.executeCommand("java.project.import");
```
Comment (rgrunber) [inline src/standardLanguageClient.ts:250]: Can we use `IMPORT_PROJECTS` from `commands.ts`. For this and the one below.

**Review by rgrunber — APPROVED** [review]
The change works well for me. Just one small thing to fix and something to confirm.

File: src/standardLanguageClient.ts [hunk 3]
```diff
@@ -217,6 +240,44 @@ export class StandardLanguageClient {
...
+			} else if (choice.startsWith(UPGRADE_GRADLE)) {
+				const useWrapper = workspace.getConfiguration().get<boolean>("java.import.gradle.wrapper.enabled");
+				if (!useWrapper) {
+					await workspace.getConfiguration().update("java.import.gradle.wrapper.enabled", true, ConfigurationTarget.Workspace);
+				}
+				const result = await window.withProgress({
+					location: ProgressLocation.Window,
```
[the remark this answers is not in the document]
Comment (testforstephen) [reply src/standardLanguageClient.ts:257]: +1

I didn't notice the status bar progress at first glance either. Notification will be more visible for me.

[the remark this answers is not in the document]
Comment (CsCherrYY) [reply src/standardLanguageClient.ts:257]: Agree that `Notification` would be better. I'll add a commit then.

**Comment by rgrunber** [comment]
Prior to merging, I would just mention https://github.com/redhat-developer/vscode-java/issues/2225 in the commit so this closes the issue.

**Pull request merged**